newIDE/app/src/MainFrame/index.js

```diff
@@ -1790,9 +1843,11 @@ class MainFrame extends React.Component<Props, State> {
               getCurrentTabIndex(this.state.editorTabs) === id;
             return (
               <ClosableTab
-                label={editorTab.name}
+                label={editorTab.label}
+                type={editorTab.type}
                 key={editorTab.key}
                 active={isCurrentTab}
+                onSetPreview={() => this._setLayoutForPreview(editorTab.label)}
```

Comment:
I know you've done it but for now I'm not a big fan of the right click menu on the tabs in this current state. We could do it, but in this case we have to pass more information about the layout/external layout/external events/extension (because the label here is very fragile, any extra character will break it).

At least it's I think more complicated than just passing a type.